ccpp/driver/GFS_diagnostics.F90

```diff
@@ -2186,6 +2186,61 @@ subroutine GFS_externaldiag_populate (ExtDiag, Model, Statein, Stateout, Sfcprop
       enddo
     endif
 
+    idx = idx + 1
```

Comment:
This entire section 2189-2243 must be in an `if` section that is only executed if those variables are allocated

Reply:
Fixed!  Thanks for catching this.